insitu/views/base.py

```diff
@@ -56,18 +57,62 @@ def ordering(self, qs):
             return qs.order_by(*order)
         return qs
 
-    def filter_queryset(self, qs):
-        for filter in self.filters:
-            value = self.request.GET.get(filter)
+    def filter_queryset(self, search):
+        """
+        Where `search` is a django_elasticsearch_dsl.search.Search object.
+        """
+        for filter_ in self.filters:
+            value = self.request.GET.get(filter_)
             if not value or value == ALL_OPTIONS_LABEL:
                 continue
-            qs = qs.filter('term', **{filter: value})
+            search = search.query('term', **{filter_: value})
 
         search_text = self.request.GET.get('search[value]', '')
-        if not search_text:
-            return qs
-        return qs.query('query_string', default_field='name',
-                        query='"' + search_text + '"')
+        if search_text:
+            search = search.query(
+                'query_string', default_field='name',
+                query='"' + search_text + '"'
+            )
+
+        if (
+                search.count() > settings.MAX_RESULT_WINDOW or not
+                hasattr(self, 'filter_fields')):
+            # If there are more than MAX_RESULT_WINDOW matching objects in the
+            # database, don't bother syncing the filter options. It would be too
+            # complicated and costly.
+            return search
+
+        search = search[0:settings.MAX_RESULT_WINDOW]
+        qs = search.to_queryset()  # If there are ever more than 10,000
```

Comment:
This throws me an error " 'Search' object has no attribute 'to_queryset'

Reply:
I changed the `requirements.txt` file, the dependencies have to be installed again (more specifically, we need `django-elasticsearch-dsl==0.4.4`).